sdk/core/src/policies/transport.rs

```diff
@@ -0,0 +1,42 @@
+use crate::policies::{Policy, PolicyResult};
+use crate::{Context, HttpClient, Request, Response};
+use std::sync::Arc;
+
+#[derive(Debug, Clone)]
+pub struct TransportOptions {
+    http_client: Arc<dyn HttpClient>,
+}
+
+impl TransportOptions {
+    pub fn new(http_client: Arc<dyn HttpClient>) -> Self {
+        Self { http_client }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct TransportPolicy {
```

Comment:
This should be used for a property on ClientOptions for a user (e.g. test infrastructure) to override.